crates/async-std-resolver/src/tests.rs

```diff
@@ -35,176 +35,116 @@ fn test_send_sync() {
     assert!(is_send_t::<LookupFuture<GenericConnection>>());
 }
 
-#[test]
-fn test_lookup_google() {
-    use testing::lookup_test;
-    let io_loop = AsyncStdConnectionProvider::new();
-
-    lookup_test::<AsyncStdConnectionProvider, AsyncStdConnectionProvider>(
-        ResolverConfig::google(),
-        io_loop.clone(),
-        io_loop,
-    )
+#[async_std::test]
+async fn test_lookup_google() {
+    subscribe();
+    lookup_test(ResolverConfig::google(), AsyncStdConnectionProvider::new()).await;
 }
 
-#[test]
-fn test_lookup_cloudflare() {
-    use testing::lookup_test;
-    let io_loop = AsyncStdConnectionProvider::new();
-    lookup_test::<AsyncStdConnectionProvider, AsyncStdConnectionProvider>(
+#[async_std::test]
+async fn test_lookup_cloudflare() {
+    subscribe();
+    lookup_test(
         ResolverConfig::cloudflare(),
-        io_loop.clone(),
-        io_loop,
+        AsyncStdConnectionProvider::new(),
     )
+    .await;
 }
 
-#[test]
-fn test_ip_lookup() {
-    use testing::ip_lookup_test;
-    let io_loop = AsyncStdConnectionProvider::new();
-    ip_lookup_test::<AsyncStdConnectionProvider, AsyncStdConnectionProvider>(
-        io_loop.clone(),
-        io_loop,
-    )
+#[async_std::test]
+async fn test_ip_lookup() {
+    subscribe();
+    ip_lookup_test(AsyncStdConnectionProvider::new()).await;
 }
 
 #[test]
 fn test_ip_lookup_across_threads() {
-    use testing::ip_lookup_across_threads_test;
-    let io_loop = AsyncStdConnectionProvider::new();
-    ip_lookup_across_threads_test::<AsyncStdConnectionProvider, AsyncStdConnectionProvider>(io_loop)
+    subscribe();
+    ip_lookup_across_threads_test::<AsyncStdConnectionProvider, AsyncStdConnectionProvider>(
```

Comment:
Nit: unlike some of these other tests, this retains the turbofished specification of the connection provider, which I think could be inferred?

Reply:
This still needs one explicit turbofish type parameter, in order to select an implementation of `Executor`. This can't be inferred since no runtime is passed in using that type.